Source/CarthageKit/Project.swift

```diff
@@ -353,34 +353,77 @@ public final class Project {
 			.map(ResolvedCartfile.init)
 	}
 	
+	/// Attempts to determine the latest version (whether satisfiable or not)
+	/// of the project's Carthage dependencies.
+	///
+	/// This will fetch dependency repositories as necessary, but will not check
+	/// them out into the project's working directory.
+	private func latestDependencies() -> SignalProducer<[Dependency<SemanticVersion>], CarthageError> {
+		return loadResolvedCartfile()
+			.map {
+				return SignalProducer<Dependency<PinnedVersion>, CarthageError>(values: $0.dependencies)
+					.map {dependency in
+						return dependency.project
+					}
+			}
+			.flatten(.Merge)
+			.map { (projectIdentifier: ProjectIdentifier) -> SignalProducer<Dependency<SemanticVersion>, CarthageError> in
+				let versions = self.versionsForProject(projectIdentifier).collect().map { (pinnedVersions: [PinnedVersion]) -> [SemanticVersion] in
```

Comment:
This should be formatted according to our conventions, where each new SignalProducer operator is on a new line.

```swift
let versions = self.versionsForProject(projectIdentifier)
    .collect()
    .map { (pinnedVersions: [PinnedVersion]) -> [SemanticVersion] in
        ..
    }
```